### PR TITLE
soroban-rpc: preflight: remove key sizes from read_bytes computation

### PR DESCRIPTION
**Pull request by 2opremio**
Core stopped including including them

Followup from https://github.com/stellar/soroban-tools/pull/913#discussion_r1313608612

### CODE REVIEW

**Review by tsachiherman — APPROVED** [review]

**Comment by 2opremio** [comment]
This has unveiled yet another `ResourceLimitExceeded` error when running the `BumpFootprint` operation. The test fails at https://github.com/2opremio/soroban-cli/blob/remove-keys-from-fee-estimation/cmd/soroban-rpc/internal/test/simulate_transaction_test.go#L748

envelope: `AAAAAgAAAABzdv3ojkzWHMD7KUoXhrPx0GH18vHKV0ZfqpMiEblG1gAAZqAAAAAAAAAABAAAAAEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAEAAAAAAAAAGQAAAAAAAAAUAAAAAQAAAAAAAAABAAAABgAAAAFnrAIfzK644FciM7c4L1q7oxCN5UdDC/Y0lL2onFdNaAAAAA8AAAAHQ09VTlRFUgAAAAABAAAAAAAAAAAAAAB0AAAAAAAAAAAAAE5FAAAAARG5RtYAAABAGAKEmnKibE8ikaCWwDf34EbFh3OeWSWc4gnsF0RY8B8+xLuOsB/EgHjf1Ns1/r0WssqDrf3P8zO2H3V9k/8pAA==`

The soroban transaction data is the following. My guess is that we are not accounting for the written expiration entries. I will try to fix that. 
Screenshot 2023-09-04 at 19 01 00

**Comment by 2opremio** [comment]
Unfortunately, adding the write bytes didn't make the test pass. Here is the new soroban transaction data:

Screenshot 2023-09-04 at 19 15 48

**Comment by 2opremio** [comment]
Now, thinking about it, the write_bytes couldn't be the problem since they were already 0 before this PR.

**Comment by 2opremio** [comment]
If you set `read_bytes` to `128` the test passes but it fails if set to `127` so we are missing `128 - 116 = 12` bytes which I am not sure how to account for.

**Comment by 2opremio** [comment]
The serialized length of an expiration ledger entry is 48, and not 36. This explains the missing 12 bytes :S 

I noticed that `fees.rs` in `rs-soroban-env` also gets it wrong (@sisuresh I guess you copy-pased it from there) so it should be corrected.

**Comment by 2opremio** [comment]
To be more specific. This prints 48:

```
	entry2 := xdr.LedgerEntry{
		LastModifiedLedgerSeq: 4,
		Data: xdr.LedgerEntryData{
			Type: xdr.LedgerEntryTypeExpiration,
			Expiration: &xdr.ExpirationEntry{
				KeyHash:             xdr.Hash{},
				ExpirationLedgerSeq: 0,
			},
		},
		Ext: xdr.LedgerEntryExt{},
	}
	bin, err := entry.MarshalBinary()
	assert.NoError(t, err)
	fmt.Println(len(bin))
```

File: cmd/soroban-rpc/lib/preflight/src/fees.rs [hunk 1]
```diff
@@ -23,9 +23,9 @@ use state_expiration::{get_restored_ledger_sequence, ExpirableLedgerEntry};
 use std::cmp::max;
 use std::convert::{TryFrom, TryInto};
 
-/// Estimate for any `ExpirationEntry` ledger entry, consisting of a 32-byte
-/// hash of the corresponding entry and 4 bytes for expiration ledger.
-const EXPIRATION_ENTRY_SIZE: u32 = 32 + 4;
+// TODO: this should be imported from soroban_env_host::fees instead
+/// Serialize XDR size for any `ExpirationEntry` ledger entry.
+const EXPIRATION_ENTRY_SIZE: u32 = 48;
```
Comment (tsachiherman) [inline cmd/soroban-rpc/lib/preflight/src/fees.rs:28]: It would be nice to have a unit test that validate that magic number - but not a blocker.

**Review by tsachiherman — APPROVED** [review]

File: cmd/soroban-rpc/lib/preflight/src/fees.rs (hunk 1 above)
Comment (2opremio) [reply cmd/soroban-rpc/lib/preflight/src/fees.rs:28]: That should be in the upstream library, no here (check the TODO)

Comment (2opremio) [reply cmd/soroban-rpc/lib/preflight/src/fees.rs:28]: I added a test at https://github.com/stellar/rs-soroban-env/pull/1034

**Comment by 2opremio** [comment]
This is ready to merge, but let's not do it until @sisuresh / @dmkozh confirm  that the `EXPIRATION_ENTRY_SIZE` fix is correct.

**Pull request merged**